astropy/coordinates/tests/test_representation.py

```diff
@@ -1046,6 +1049,16 @@ class LogDRepresentation(BaseRepresentation):
                                         ('lat', Latitude),
                                         ('logr', u.Dex)])
 
+    # now try passing in a differential
+    diff = CartesianDifferential(d_x=1 * u.km/u.s,
```

Comment:
The problem is in part that your unit vectors are wrong - they should be always have unit length, so no need to include the `d` factor (they should be identical to `SphericalRepresenation.unit_vectors`, so you could just link to that).

For the scale factor, it should just be `1.` -- note that this makes tricky unit verification, the `d^n<comp>/dt^2` implies a difference of `<comp>` which for `LogDRepresentation` does not just have the unit of `log_d`... I'd say we postpone this to later...